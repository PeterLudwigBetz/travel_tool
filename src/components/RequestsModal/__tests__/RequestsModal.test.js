import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import configureStore from 'redux-mock-store';
import ConnectedRequestDetailsModal,
{ RequestDetailsModal } from '../RequestsModal';
import { fetchRequestsDetailsResponse,
  userImage } from '../../../redux/__mocks__/mocks';
import requestData from '../__mocks__/requestData';

let wrapper;
const props = {
  requestId: 'xDh20cuGx',
  user: {
    picture: 'http://my-image'
  },
  requestData,
  fetchUserRequestDetails: sinon.spy(() => Promise.resolve()),
  requests: {
    requestData: {
      id: '1',
      name: 'Orion',
      status: 'Approved'
    }
  },
  updateRequestStatus: jest.fn(() => Promise.resolve()),
  isStatusUpdating: false
};
const componentDidMountSpy = sinon.spy(RequestDetailsModal.prototype, 'componentDidMount');
describe('Render RequestsModal component', () => {
  beforeEach(() => {
    wrapper = shallow(<RequestDetailsModal {...props} />);
  });
  // make our assertions and what we expect to happen
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('tests that the component successfully rendered', () => {
    expect(wrapper.find('div').exists()).toBeTruthy();
  });

  it('should call componentDidMount()', () => {
    const {fetchUserRequestDetails, requestData} =  props;
    expect(componentDidMountSpy.called).toBe(true);
    expect(fetchUserRequestDetails.called).toEqual(true);
    expect(fetchUserRequestDetails.calledWith(requestData.id)).toEqual(true); 
  });

  it('should render the RequestsModal as expected', () => {
    expect(wrapper.length).toBe(1);
  });

  it('should render approveTextColor when button is clicked', () => {
    const approveButton = wrapper.find('#b1').at(0);
    approveButton.simulate('click');
    expect(wrapper.state('approveTextColor')).toBe('white');
  });

  it('should render rejectTextColor when button is clicked', () => {
    const rejectButton = wrapper.find('#b2').at(0);
    rejectButton.simulate('click');
    expect(wrapper.state('rejectTextColor')).toBe('white');
  });

  it('should render approveColor when button is clicked', () => {
    const approveButton = wrapper.find('#b1').at(0);
    approveButton.simulate('click');
    expect(wrapper.state('approveColor')).toBe('#49AAAF');
  });

  it('should render rejectColor when button is clicked', () => {
    const rejectButton = wrapper.find('#b2').at(0);
    rejectButton.simulate('click');
    expect(wrapper.state('rejectColor')).toBe('#FF5359');
  });
  const initialState = {};
  const mockStore = configureStore();
  const store = mockStore(initialState);

  it('should call renderStatusAsBadge function', () => {
    const action = wrapper.instance();
    const renderStatusAsBadge = jest.spyOn(wrapper.instance(), 'renderStatusAsBadge');
    action.renderStatusAsBadge('Approved');
    expect(renderStatusAsBadge).toBeCalled();
  });

  it('should call changeButtonColor function', () => {
    const button = {
      id: 1,
    };
    const action = wrapper.instance();
    const changeButtonColor = jest.spyOn(wrapper.instance(), 'changeButtonColor');
    action.changeButtonColor(button, 'Rejected');
    expect(changeButtonColor.mock.calls.length).toBe(1);
  });

  describe('Connected RequestDetailsModal component', () => {
    it('tests that the component successfully rendered', () => {
      const store = mockStore({
        requests: fetchRequestsDetailsResponse.requestData,
        auth: { ...userImage },
        approvals: [],
        modal: {modal: {}}
      });
      const wrapper = shallow(<ConnectedRequestDetailsModal store={store} />);
      expect(wrapper.length).toBe(1);
    });
  });
});
