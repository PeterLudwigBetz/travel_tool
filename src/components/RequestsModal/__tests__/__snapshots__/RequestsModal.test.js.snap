// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render RequestsModal component should match snapshot 1`] = `
<Fragment>
  <div>
    <div
      className="modal__user-info"
    >
      <ImageLink
        altText="avatar"
        imageClass="modal__oval"
        imageSrc="http://my-image"
      />
      <span
        className="modal__text-size"
      >
        Test user C
      </span>
      <div
        className="modal__modal3"
      >
        Software Developer
        ,
        TDD
      </div>
    </div>
  </div>
  <div
    className="modal__travel-place"
  >
    <span
      className="modal__dialog1"
    >
      Request to travel to:
      <span
        className="modal__city-name"
      >
        Nairobi
      </span>
    </span>
    <span
      className="modal__dialog-from"
    >
      From:
      <span
        className="modal__city-name"
      >
        Lagos
      </span>
    </span>
  </div>
  <div
    className="modal__modal-date"
  >
    <TravelLink
      divClass="modal__travel-date"
      dynamicDate="04 Sep 2018"
      dynamicText="Date submitted"
      innerClass="modal__travel-dates"
      nextClass="modal__date-text"
    />
    <TravelLink
      divClass="modal__travel-date"
      dynamicDate="12 Sep 2018"
      dynamicText="Target depature date"
      innerClass="modal__travel-dates"
      nextClass="modal__date-text"
    />
    <TravelLink
      divClass="modal__travel-date"
      dynamicDate="12 Nov 2018"
      dynamicText="Target return date"
      innerClass="modal__travel-dates"
      nextClass="modal__date-text"
    />
  </div>
  <div
    className="modal__button-below"
  >
    <span
      key="1"
    >
      <span
        className="modal__dialog-btn"
      >
        <button
          className="modal__button-submitted-text bg"
          disabled={true}
          id="b1"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
              "color": "",
            }
          }
          type="button"
        >
          Approve
        </button>
      </span>
    </span>
    <span
      key="2"
    >
      <span
        className="modal__dialog-btn"
      >
        <button
          className="modal__button-rejected-text"
          disabled={true}
          id="b2"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "",
              "color": "",
            }
          }
          type="button"
        >
          Reject
        </button>
      </span>
    </span>
  </div>
  <AddComment />
  <Connect(CommentBox)
    requestId="xDh20cuGx"
  />
  <div
    className="modal__modal1"
  >
    <span
      className="modal__mdl-icons"
    >
      <ImageLink
        altText="avatar"
        imageClass="modal__oval-copy"
        imageSrc="Oval2.png"
      />
      <span
        className="modal__user-name"
      >
        Samuel Kubai
      </span>
      <span
        className="modal__approval-status"
      >
        approved your travel request.
      </span>
      <span
        className="modal__hours-status"
      >
        5 hours ago
      </span>
    </span>
  </div>
  <div
    id="comments"
  >
    <div
      className="modal__modal1"
      key="2"
    >
      <hr />
      <div
        className="modal__mdl-icons"
      >
        <ImageLink
          altText="avatar"
          imageClass="modal__oval-copy"
          imageSrc="http://my-image"
        />
        <span
          className="modal__user-name"
        >
          Random User Two
        </span>
        <span
          className="modal__hours-status"
        >
          in 22 days
        </span>
        <span
          className="modal__dialog"
        >
          <button
            className="modal__delete-btn"
            type="button"
          >
            Delete
          </button>
        </span>
        <div
          className="modal__modal2"
        >
          <div
            className="modal__status-update"
            dangerouslySetInnerHTML={
              Object {
                "__html": "This is another comment",
              }
            }
          />
           
        </div>
      </div>
    </div>
    <div
      className="modal__modal1"
      key="1"
    >
      <hr />
      <div
        className="modal__mdl-icons"
      >
        <ImageLink
          altText="avatar"
          imageClass="modal__oval-copy"
          imageSrc="http://my-image"
        />
        <span
          className="modal__user-name"
        >
          Random User
        </span>
        <span
          className="modal__hours-status"
        >
          8 days ago
        </span>
        <span
          className="modal__dialog"
        >
          <button
            className="modal__delete-btn"
            type="button"
          >
            Delete
          </button>
        </span>
        <div
          className="modal__modal2"
        >
          <div
            className="modal__status-update"
            dangerouslySetInnerHTML={
              Object {
                "__html": "This is a comment",
              }
            }
          />
           
        </div>
      </div>
    </div>
  </div>
</Fragment>
`;
